src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRevocationListBuilder.cs

```diff
@@ -0,0 +1,327 @@
+        /// </summary>
+        /// <param name="serialNumber">
+        ///   The serial number of the certificate to revoke.
+        /// </param>
+        /// <param name="revocationTime">
+        ///   The time the certificate was revoked,
+        ///   or <see langword="null" /> to use the current system time.
+        ///   The default is <see langword="null" />.
+        /// </param>
+        /// <param name="reason">
+        ///   The reason why the certificate was revoked,
+        ///   or <see langword="null" /> to not include a reason.
+        ///   The default is <see langword="null" />.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="serialNumber"/> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="reason" /> is not a supported revocation reason.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="serialNumber"/> is empty.
+        /// </exception>
+        public void AddEntry(
+            byte[] serialNumber,
+            DateTimeOffset? revocationTime = null,
+            X509RevocationReason? reason = null)
+        {
+            ArgumentNullException.ThrowIfNull(serialNumber);
+
+            AddEntry(new ReadOnlySpan<byte>(serialNumber), revocationTime, reason);
+        }
+
+        /// <summary>
+        ///   Adds the specified serial number to the revocation list with an optional revocation time
+        ///   and an optional revocation reason.
+        /// </summary>
+        /// <param name="serialNumber">
+        ///   The serial number of the certificate to revoke.
+        /// </param>
+        /// <param name="revocationTime">
+        ///   The time the certificate was revoked,
+        ///   or <see langword="null" /> to use the current system time.
+        ///   The default is <see langword="null" />.
+        /// </param>
+        /// <param name="reason">
+        ///   The reason why the certificate was revoked,
+        ///   or <see langword="null" /> to not include a reason.
+        ///   The default is <see langword="null" />.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="reason" /> is not a supported revocation reason.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="serialNumber"/> is empty.
+        /// </exception>
+        public void AddEntry(
+            ReadOnlySpan<byte> serialNumber,
+            DateTimeOffset? revocationTime = null,
+            X509RevocationReason? reason = null)
+        {
+            if (serialNumber.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptyOrNullArray, nameof(serialNumber));
+
+            if (serialNumber.Length > 1)
+            {
+                if ((serialNumber[0] == 0x00 && serialNumber[1] < 0x80) ||
```

Comment:
Similar to our discussion in the AKID extension... do we care about the well-formedness of the serial number here? If someone wants to revoke a cert with a bad serial number (a pretty good reason to revoke it), should they be able to?

Reply:
It looks like this permits negative serial numbers, so I think this is just checking for a serial number that is not a valid DER integer, so maybe it's fine.